app/models/topic.rb

```diff
@@ -41,8 +41,15 @@ class Topic < ActiveRecord::Base
 
   acts_as_readable
 
-  def self.recent_topics
-    find_by_sql %q{
+  def self.recent_topics user=nil
+    constraint = ""
+
+    if user && user.groups.any?
+      forumer_ids = user.groups.map(&:forumers).flatten.map(&:id)
+      constraint = "OR forumers.id IN (#{forumer_ids.join(",")})" unless forumer_ids.empty?
+    end
+
+    find_by_sql %Q(
```

Comment:
Use `%` instead of `%Q`.<br>Use `%Q` only for strings that contain both single quotes and double quotes, or for dynamic strings that contain double quotes.